### PR TITLE
docs(update) Clarify support for Jakarta EL

### PR DESCRIPTION
**Pull request by venetrius**
related to: camunda/camunda-bpm-platform/issues/3992

### CODE REVIEW

File: content/user-guide/process-engine/expression-language.md [hunk 1]
```diff
@@ -10,9 +10,11 @@ menu:
 
 ---
 
-Camunda 7 supports Unified Expression Language (EL), specified by the [Jakarta Expression
-Language 4.0 standard][JakartaEL]. To do so, it maintains a custom version of the open source 
-[JUEL][] implementation. To get more general information about the usage of Expression Language, 
+Camunda 7 supports the Unified Expression Language (EL), specified by the [Jakarta Expression
+Language 4.0 standard][JakartaEL]. To do so, it maintains a custom version of the open source [JUEL][] implementation.
+Note, this JUEL implementation does not support collection constructions in expressions *(`${...}`)*.
```
Comment (yanavasileva) [inline content/user-guide/process-engine/expression-language.md:15]: 💭 I think there are more features that are not supported: lambda, enums for example. Can we re-write it to include what is supported or list all of what is not supported? 
Resources:
* https://juel.sourceforge.net/
* https://github.com/beckchr/juel/tree/master/modules/impl/src/main/java/de/odysseus/el/tree/impl

Comment (venetrius) [reply content/user-guide/process-engine/expression-language.md]: Good point, added the list of unsupported expressions

File: content/user-guide/process-engine/expression-language.md [hunk 2]
```diff
@@ -10,9 +10,21 @@ menu:
...
+Language 4.0 standard][JakartaEL]. To do so, it maintains a custom version of the open source [JUEL][] implementation.
+
+Note, compared to EL 4.0 this JUEL implementation has the following limitations: 
+
+
+- Initializing collections directly within expressions (e.g., `${[1,2,3]}`) is NOT supported.
+
+- Lambda expressions (e.g., inline functions `${((x,y)->x+y)(3,4)}`) are NOT supported.
+
+- Referencing static fields (e.g., `${Boolean.TRUE}` ), static functions (e.g., `${Integer.parseInt("123")}`) and enums (e.g., `${Thread.State.TERMINATED}` ) are NOT supported.
```
Comment (yanavasileva) [inline content/user-guide/process-engine/expression-language.md:23]: 🙃 Spaces and punctuation:
```suggestion
- Referencing static fields (e.g., `${Boolean.TRUE}`), static functions (e.g., `${Integer.parseInt("123")}`), and enums (e.g., `${Thread.State.TERMINATED}`) are NOT supported. 
```

File: content/user-guide/process-engine/expression-language.md [hunk 3]
```diff
@@ -10,9 +10,21 @@ menu:
...
+Note, compared to EL 4.0 this JUEL implementation has the following limitations: 
+
+
+- Initializing collections directly within expressions (e.g., `${[1,2,3]}`) is NOT supported.
+
+- Lambda expressions (e.g., inline functions `${((x,y)->x+y)(3,4)}`) are NOT supported.
+
+- Referencing static fields (e.g., `${Boolean.TRUE}` ), static functions (e.g., `${Integer.parseInt("123")}`) and enums (e.g., `${Thread.State.TERMINATED}` ) are NOT supported. 
+
+- The Assignment Operator `(A=B)`, the String Concatenation Operator `(A+=B)` and the Semicolon Operator `(A ; B)` are NOT supported.
```
Comment (yanavasileva) [inline content/user-guide/process-engine/expression-language.md:25]: 🙃 Missing comma:
```suggestion
- The Assignment Operator `(A=B)`, the String Concatenation Operator `(A+=B)`, and the Semicolon Operator `(A ; B)` are NOT supported.
```

**Review by yanavasileva — APPROVED** [review]
👍 Thank you for considering my feedback. Small remark about missing commas below.

**Pull request merged**